app/controllers/reports_controller.rb

```diff
@@ -23,6 +23,11 @@ def create
 
   def new
     @info_request = InfoRequest.find_by_url_title!(params[:request_id])
+    if params[:comment_id]
+      if @info_request.comments.map(&:id).include?(params[:comment_id].to_i)
```

Comment:
Good spot, thanks - had overlooked that a non-matching id would safely return `nil` here